packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -115,7 +115,13 @@ bool ArbCore::machineIdle() {
 ArbCore::message_status_enum ArbCore::messagesStatus() {
     auto current_status = message_data_status.load();
     if (current_status == MESSAGES_SUCCESS) {
-        message_data_status = MESSAGES_EMPTY;
+        auto success = message_data_status.compare_exchange_strong(
+            current_status, MESSAGES_EMPTY);
+        if (!success) {
+            // Status was not updated because value changed, so return current
+            // value
+            current_status = message_data_status.load();
```

Comment:
In theory, maybe the outer if statement should be a while loop? In practice it's unlikely to happen multiple times though

Reply:
It is already in a loop on the Go side